gsrs-module-clinical-trials-spring-boot-autoconfigure/src/main/java/gov/hhs/gsrs/clinicaltrial/us/validators/GeneralTrialOnUpdateValidator.java

```diff
@@ -28,80 +28,75 @@ public boolean supports(ClinicalTrialUS newValue, ClinicalTrialUS oldValue, Vali
     final String newLastModifiedDateNullErrorTemplate = "New Last Modified Date is null.";
     final String oldCreationDateNullErrorTemplate = "Old Creation Date is null.";
     final String oldLastModifiedDateNullErrorTemplate = "Old Last Modified Date is null.";
-    final String newOldLastModifiedDatesDifferentErrorTemplate = "New and old Last Modified Dates must be equal.";
-
-
-    // final String newCreationDateDifferentFromOldErrorTemplate = "New Creation Date different that record being replaced.";
-    // final String newLastModifiedDateDifferentFromOldErrorTemplate = "New Last Modified Date different that record being replaced.";
-
+    final String newOldCreationDatesDifferentErrorTemplate = "New and old Creation Dates must be equal in record to be updated.";
+    final String newOldLastModifiedDatesDifferentErrorTemplate = "New and old Last Modified Dates must be equal in record to be updated.";
 
     final Pattern trialNumberPattern = Pattern.compile("^NCT[\\d]+$");
 
     @Override
     public void validate(ClinicalTrialUS objnew, ClinicalTrialUS objold, ValidatorCallback callback) {
-        System.out.println("Inside GeneralTrialOnUpdateValidator");
-
-        String trialNumber = objnew.getTrialNumber();
-        if(trialNumber==null) {
-            callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(trialNumberNullErrorTemplate)));
-        }
-        boolean formatOK = trialNumberPattern.matcher(trialNumber).matches();
-        if(!formatOK) {
-            callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(badlyFormattedTrialNumberTemplate, trialNumber)));
-        }
-        System.out.println("Checking that trial already exists");
-        // ask danny about this? how is objold populated.
-        if(objold != null) {
-            System.out.println("Checking that trial already exists: " + objold.getTrialNumber());
-            System.out.println("Old Modified date: " + objold.getLastModifiedDate());
-        }
-        Optional<ClinicalTrialUS> found = repository.findById(objnew.getTrialNumber());
-        if(!found.isPresent()) {
-            callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(trialNumberShouldAlreadyExistErrorTemplate, trialNumber)));
+        if (objold == null) {
+            callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(objoldIsNullErrorTemplate)));
+        } else if (objnew == null) {
+            callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(objnewIsNullErrorTemplate)));
         } else {
-            ClinicalTrialUS ct = found.get();
+            String trialNumber = objnew.getTrialNumber();
+            if (trialNumber == null) {
+                callback.addMessage(GinasProcessingMessage.ERROR_MESSAGE(String.format(trialNumberNullErrorTemplate)));
+            }
+            boolean formatOK = trialNumberPattern.matcher(trialNumber).matches();
```

Comment:
Move this and the things below into an else from above, otherwise you may get a NPE, which will make the validation harder and may result in a 500 instead of a validation error.